controllers/topolvm_node.go

```diff
@@ -0,0 +1,359 @@
+
+	// Affinity and tolerations
+	nodeSelector, tolerations := extractNodeSelectorAndTolerations(*lvmCluster)
+	topolvmNodeAffinity := &corev1.Affinity{}
+	if nodeSelector != nil {
+		topolvmNodeAffinity = &corev1.Affinity{
+			NodeAffinity: &corev1.NodeAffinity{RequiredDuringSchedulingIgnoredDuringExecution: nodeSelector},
+		}
+	}
+	topolvmNodeTolerations := []corev1.Toleration{{Operator: corev1.TolerationOpExists}}
+	if tolerations != nil {
+		topolvmNodeTolerations = tolerations
+	}
+
+	nodeDaemonSet := &v1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      TopolvmNodeDaemonsetName,
+			Namespace: lvmCluster.Namespace,
+		},
+		Spec: v1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app.kubernetes.io/name": lvmCluster.Name,
+				},
+			},
+			UpdateStrategy: v1.DaemonSetUpdateStrategy{
+				Type: v1.RollingUpdateDaemonSetStrategyType,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: lvmCluster.Name,
+					Labels: map[string]string{
+						"app.kubernetes.io/name": lvmCluster.Name,
+					},
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: TopolvmNodeServiceAccount,
+					InitContainers:     iContainers,
+					Containers:         containers,
+					Volumes:            volumes,
+					HostPID:            true,
+					Tolerations:        topolvmNodeTolerations,
+					Affinity:           topolvmNodeAffinity,
+				},
+			},
+		},
+	}
+
+	return nodeDaemonSet
+}
+func getNodeInitContainer() *corev1.Container {
+	command := []string{
+		"sh",
+		"-c",
+		fmt.Sprintf("until [ -f %s ]; do echo waiting for lvmd config file; sleep 5; done", lvmdConfigFile),
+	}
+
+	volumeMounts := []corev1.VolumeMount{
+		{Name: "lvmd-config-dir", MountPath: "/etc/topolvm"}}
+
+	fileChecker := &corev1.Container{
+		Name:         "file-checker",
+		Image:        auxImage,
+		Command:      command,
+		VolumeMounts: volumeMounts,
+	}
+
+	return fileChecker
+}
+func getLvmdContainer() *corev1.Container {
+	command := []string{
+		"/lvmd",
+		fmt.Sprintf("--config=%s", lvmdConfigFile),
+		"--container=true",
+	}
+
+	resourceRequirements := corev1.ResourceRequirements{
+		Limits: corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse(TopolvmNodeCPULimit),
+			corev1.ResourceMemory: resource.MustParse(TopolvmNodeMemLimit),
+		},
+		Requests: corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse(TopolvmNodeCPURequest),
+			corev1.ResourceMemory: resource.MustParse(TopolvmNodeMemRequest),
+		},
+	}
+
+	volumeMounts := []corev1.VolumeMount{
+		{Name: "lvmd-socket-dir", MountPath: "/run/topolvm"},
+		{Name: "lvmd-config-dir", MountPath: "/etc/topolvm"}}
+
+	privilege := true
+	runUser := int64(0)
+	lvmd := &corev1.Container{
+		Name:  "lvmd",
+		Image: TopolvmCsiImage,
+		SecurityContext: &corev1.SecurityContext{
+			Privileged: &privilege,
```

Comment:
- you are using `privilege` here and in `topolvm-node` scc privileged isn't set to `true` so, constraint isn't allowing this pod to be scheduled
- eventually, we need to remove privilege usage across every component as not all components need this.

Reply:
This might be a clue about what the "topolvm-node" scc is not working, i will address this after we merge the first version.